pennylane/fourier/visualization.py

```diff
@@ -0,0 +1,606 @@
+            be a pair of axes from a subplot where ``sharex="row"`` and ``sharey="col"``.
+        colour_dict (dict[str, str]): A dictionary of the form {"real" : colour_string,
+            "imag" : other_colour_string} indicating which colours should be used in the plot.
+        show_freqs (bool): Whether or not to print the frequency labels on the plot axis.
+
+    Returns:
+        ax: The axes on which the data is plotted.
+
+    **Example**
+
+    .. code-block:: python
+
+        import matplotlib as plt
+        from pennylane.fourier import fourier_coefficients, fourier_violin_plot
+
+        f = ... # A function
+        n_inputs = ... # Number of inputs to the function
+        degree = ... # Degree to which coefficients should be calculated
+
+        # Calculate the Fourier coefficients; may be a single set or a list of
+        # multiple sets of coefficients
+        coeffs = fourier_coefficients(f, n_inputs, degree)
+
+        # Set up subplots and plot
+        fig, ax = plt.subplots(2, 1, sharey=True, figsize=(15, 4))
+        fourier_violin_plot(coeffs, n_inputs, ax, show_freqs=True);
+
+    """
+    # Check dimensionality; it's possible a user has provided only a single set
+    # of coefficients to this function.
+    if len(coeffs.shape) == n_inputs:
+        coeffs = np.array([coeffs])
+
+    # The axis received must be a pair of axes in a subplot.
+    if colour_dict is None:
+        colour_dict = {"real": "purple", "imag": "green"}
+
+    # extract the x ticks
+    nvecs = list(to_dict(coeffs[0]).keys())
+    nvecs_formatted = [format_nvec(nvec) for nvec in nvecs]
+
+    # make data
+    data = {}
+    data["real"] = np.array([[c[nvec].real for nvec in nvecs] for c in coeffs])
+    data["imag"] = np.array([[c[nvec].imag for nvec in nvecs] for c in coeffs])
+
+    for (data_type, axis) in zip(["real", "imag"], ax):
+        violin = axis.violinplot(data[data_type], showextrema=False)
+        for bd in violin["bodies"]:
+            bd.set_color(colour_dict[data_type])
+            bd.set_alpha(0.7)
+        axis.set_ylabel(data_type)
+        axis.xaxis.set_ticks(np.arange(1, len(data[data_type][0]) + 1))
+        _adjust_spine_placement(axis)
+
+    # Format axes
+    ax[0].tick_params(axis="x", colors="white")  # hack to get rid of ticks but keep grid
+
+    if show_freqs:
+        ax[1].tick_params(axis="x", which="both", length=0)  # remove ticks without removing labels
+        ax[1].xaxis.set_ticklabels(nvecs_formatted, fontsize=10, color="grey")
+        ax[1].xaxis.set_ticks_position("top")
+    else:
+        ax[1].tick_params(axis="x", colors="white")  # hack to get rid of ticks but keep grid
+
+    return ax
+
+
+def fourier_box_plot(coeffs, n_inputs, ax, colour_dict=None, show_freqs=True, show_fliers=True):
+    """Plots a set of Fourier coefficients as a box plot.
+
+    Args:
+        coeffs (array[complex]): A single set of Fourier coefficients.
```

Comment:
By the way, I always stumble over this...when I feed samples, I feel the most natural data structure is a list of samples. Do you think we could accept a list here and just parse inside the function?

Reply:
Ah! This is just a copy-paste error in the docs, thanks for catching that. This function does work with a list of samples. (It also accepts a single sample which is then processed and turned into a list down in line 146.) (The only function that explicitly takes just a single set of samples is `fourier_bar_plot`)